fs/operations/operations_test.go

```diff
@@ -847,6 +847,95 @@ func TestCopyFileBackupDir(t *testing.T) {
 	fstest.CheckItems(t, r.Fremote, file1old, file1)
 }
 
+// Test with CompareDest set
+func TestCopyFileCompareDest(t *testing.T) {
+	r := fstest.NewRun(t)
+	defer r.Finalise()
+
+	fs.Config.CompareDest = r.FremoteName + "/CompareDest"
+	defer func() {
+		fs.Config.CompareDest = ""
+	}()
+
+	file2 := r.WriteObject(context.Background(), "CompareDest/one", "one", t1)
```

Comment:
These tests didn't discover the `nil` pointer bug caused by not checking the error from `NewObject` which means that they need more in them.

I think you need another file where the file doesn't exist in `CompareDest`.

I recommend using a coverage tool so you can see you hit all the branches (or as many as you can!) in the code. Here is how to do it: https://blog.golang.org/cover

Reply:
thanks for the link, will write better coverage